From a76c19a3c9f0736087b67fdca44902708c1c083a Mon Sep 17 00:00:00 2001
From: Dmitry Shusharin <pmdvsh@gmail.com>
Date: Sat, 14 Dec 2019 17:15:21 +0700
Subject: [PATCH] x264enc: corrected em_data value in CEA-708 CC SEI message
 (fixes #28)

Section 4.4 of CEA-708-D specification (table 2) requires all bits to be
set inside em_data field. h264parse element (and possible third-party
decoders such as libav) also follows this requirement.

https://gitlab.freedesktop.org/gstreamer/gst-plugins-ugly/issues/28
---
 ext/x264/gstx264enc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/x264/gstx264enc.c b/ext/x264/gstx264enc.c
index f313b211..ee275dc8 100644
--- a/ext/x264/gstx264enc.c
+++ b/ext/x264/gstx264enc.c
@@ -2413,7 +2413,7 @@ gst_x264_enc_add_cc (GstBuffer * buffer, x264_picture_t * pic_in)
      */
     pic_in->extra_sei.payloads[i].payload[8] =
         ((cc_meta->size / 3) & 0x1f) | 0x40;
-    pic_in->extra_sei.payloads[i].payload[9] = 0;       /* 8 bits em_data, unused */
+    pic_in->extra_sei.payloads[i].payload[9] = 255;       /* 8 bits em_data, unused */
     pic_in->extra_sei.payloads[i].payload[cc_meta->size + 10] = 255;    /* 8 marker bits */
   }
 }
-- 
2.24.1


From fcddd6e184e4cae1876affe62449aa68fb7e2c8d Mon Sep 17 00:00:00 2001
From: Dmitry Shusharin <pmdvsh@gmail.com>
Date: Sat, 14 Dec 2019 18:49:54 +0700
Subject: [PATCH] x264enc: fixed codestyle

---
 ext/x264/gstx264enc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/x264/gstx264enc.c b/ext/x264/gstx264enc.c
index ee275dc8..79bbe009 100644
--- a/ext/x264/gstx264enc.c
+++ b/ext/x264/gstx264enc.c
@@ -2413,7 +2413,7 @@ gst_x264_enc_add_cc (GstBuffer * buffer, x264_picture_t * pic_in)
      */
     pic_in->extra_sei.payloads[i].payload[8] =
         ((cc_meta->size / 3) & 0x1f) | 0x40;
-    pic_in->extra_sei.payloads[i].payload[9] = 255;       /* 8 bits em_data, unused */
+    pic_in->extra_sei.payloads[i].payload[9] = 255;     /* 8 bits em_data, unused */
     pic_in->extra_sei.payloads[i].payload[cc_meta->size + 10] = 255;    /* 8 marker bits */
   }
 }
-- 
2.24.1

